### PR TITLE
negation of an undefined value should return true

### PR DESCRIPTION
**Pull request by eric-therond**
When I read this discussion https://github.com/open-policy-agent/opa/issues/1622#issuecomment-520547385
It's not clear for me what should be the value of a negated undefined value

But currently, on rego playground, `not undefined` returns `true`

### CODE REVIEW

**Review by anakrish — COMMENTED** [review]
LGTM,

Thanks for the contribution.

Can you also add the following test:
```
        package test

        import future.keywords.if

        p if false # undefined value

        q1 {
          not p # `not undefined` evaluates to true
        }
        
        q2 = [p] # q2 = [undefined] = undefined
        
        q3 {
          not [p] # `not undefined` evaluates to true
        }

        q4 {
          not q3 # `not true` making it undefined
        }
```

Rego playground behaves differently, which we might need to follow up about.

**Comment by eric-therond** [comment]
It's done

I found [this issue](https://github.com/open-policy-agent/opa/issues/1877) explaining that queries could be rewritten in REPL and produce unexpected results regarding negated expressions.

Maybe it can explain the difference?

**Comment by anakrish** [comment]
@eric-therond Thanks for finding out why OPA Rego behaves differently.

File: src/interpreter.rs [hunk 1]
```diff
@@ -760,7 +760,9 @@ impl<'source> Interpreter<'source> {
                 value,
                 collection,
             } => self.eval_some_in(span, key, value, collection, stmts)?,
-            Literal::NotExpr { expr, .. } => matches!(self.eval_expr(expr)?, Value::Bool(false)),
+            Literal::NotExpr { expr, .. } => {
```
Comment (anakrish) [inline src/interpreter.rs:763]: Add link to the discussion https://github.com/open-policy-agent/opa/issues/1622#issuecomment-520547385 in comments here.

File: tests/interpreter/cases/negation/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        x[a] {
+          not myequal(1)
+          a := "hello"
+        }
+    query: data.test
+    want_result:
+      x: 
+        set!: [hello]
+
+  - note: not-more-undefined
```
Comment (anakrish) [inline tests/interpreter/cases/negation/tests.yaml]:
```suggestion
  - note: not-more-undefined-OPA-INCOMPATIBLE
```
We could used a naming convention such as above to keep track of behavioral differences from OPA.
Also add the link to https://github.com/open-policy-agent/opa/issues/1877 in comments.

**Review by anakrish — COMMENTED** [review]
LGTM.

I have couple of suggestions.

File: tests/interpreter/cases/negation/tests.yaml (hunk 2 above)
Comment (eric-therond) [reply tests/interpreter/cases/negation/tests.yaml]: Done

**Pull request merged**